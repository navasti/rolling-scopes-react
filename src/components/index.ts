export { Loader } from './Loader';
export { Modal } from './Modal';
