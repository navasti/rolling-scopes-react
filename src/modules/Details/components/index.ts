export { MoveCardDetails } from './MoveCardDetails';
export { TypeCardDetails } from './TypeCardDetails';
export { PokemonCardDetails } from './PokemonCardDetails';
