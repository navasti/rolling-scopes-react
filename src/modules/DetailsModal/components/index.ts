export { TypeModalContent } from './TypeModalContent';
export { MoveModalContent } from './MoveModalContent';
export { PokemonModalContent } from './PokemonModalContent';
