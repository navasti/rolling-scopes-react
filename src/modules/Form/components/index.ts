export { BirthdayField } from './BirthdateField';
export { ErrorMessage } from './ErrorMessage';
export { ConsentField } from './ConsentField';
export { AvatarField } from './AvatarField';
export { GenderField } from './GenderField';
export { ShinyField } from './ShinyField';
export { TypeField } from './TypeField';
export { NameField } from './NameField';
export { FormCard } from './FormCard';
