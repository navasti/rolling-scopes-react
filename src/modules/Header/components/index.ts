export { LocationBadge } from './LocationBadge';
export { Navigation } from './Navigation';
