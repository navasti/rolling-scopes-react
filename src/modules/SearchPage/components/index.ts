export { SearchBar } from './SearchBar';
export { Cards } from './Cards';
export { Card } from './Card';
export { Tabs } from './Tabs';
