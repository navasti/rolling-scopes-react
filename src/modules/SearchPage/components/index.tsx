export { SearchBar } from './SearchBar';
export { Card } from './Card';
