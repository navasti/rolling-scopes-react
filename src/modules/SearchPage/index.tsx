import { fetchDetailedPokemons, fetchPokemons } from 'utils';
import { API_URL, INPUT_VALUE_KEY } from 'appConstants';
import { SearchBar, Card } from './components';
import React, { ChangeEvent } from 'react';
import { PokemonDetails } from 'types';
import { Loader } from 'components';
import { Layout } from 'modules';
import * as S from './styled';

type Props = {
  componentName: string;
  location: string;
};

type State = {
  pokemons: Array<PokemonDetails>;
  inputValue: string;
  isLoading: boolean;
};

export class SearchPage extends React.Component<Props, State> {
  state: State = {
    isLoading: true,
    inputValue: '',
    pokemons: [],
  };
  componentDidMount() {
    const value = window.localStorage.getItem(INPUT_VALUE_KEY);
    if (value != null) {
      this.setState({ inputValue: value });
    }
    fetchPokemons(API_URL)
      .then((pokemons) => fetchDetailedPokemons(pokemons))
      .then((pokemons) => this.setState({ pokemons }))
      .finally(() => setTimeout(() => this.setState({ isLoading: false }), 1000));
  }
  onChange = (event: ChangeEvent<HTMLInputElement>) => {
    const { value } = event.target;
    this.setState({ inputValue: value });
    window.localStorage.setItem(INPUT_VALUE_KEY, value);
  };
  render() {
    const { inputValue, isLoading, pokemons } = this.state;
    const { componentName, location } = this.props;
    return (
      <Layout componentName={componentName} location={location}>
        <S.SearchPageView>
          <SearchBar onChange={this.onChange} label="Local Storage Input" inputValue={inputValue} />
          {!isLoading && !pokemons.length ? (
            <S.CommonMessage>No pokemons found</S.CommonMessage>
          ) : (
            <S.CardsWrapper isLoading={isLoading}>
              {isLoading ? (
                <Loader />
              ) : (
                pokemons.map((pokemon) => <Card key={pokemon.id} pokemon={pokemon} />)
              )}
            </S.CardsWrapper>
          )}
        </S.SearchPageView>
      </Layout>
    );
  }
}
