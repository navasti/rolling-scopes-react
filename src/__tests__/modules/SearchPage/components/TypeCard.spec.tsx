import { screen, render } from '@testing-library/react';
import { TypeCard } from 'modules/SearchPage/components';

describe('', () => {
  it('', () => {});
});
