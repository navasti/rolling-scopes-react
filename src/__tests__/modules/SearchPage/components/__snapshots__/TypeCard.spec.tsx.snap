// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypeCard type card should be rendered and match snapshot 1`] = `
<div>
  <button
    class="sc-iqcoie iHaGah"
    data-testid="type-card"
  >
    <p>
      fighting
    </p>
  </button>
</div>
`;
